src/Assets/AssetsStorage.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "GameManager.h"
+#include "System/filesystem.h"
+#include "Assets/Assets.h"
+#include "Parsers/Parser.h"
+
+/**
+ * Just use getFile and AssetStorageKey to get what you want :)
+ */
+
+namespace Sourcehold {
+    namespace Assets {
+        using namespace std;
+
+        enum class AssetStorageKey {
+            BODY_LORD_GM1,
+            BODY_DEER_GM1,
+            
+            TREE_APPLE_GM1,
+            TREE_BIRCH_GM1,
+            TREE_CHESTNUT_GM1,
+            TREE_OAK_GM1,
+            TREE_PINE_GM1,
+            TREE_SHRUB1_GM1,
+            TREE_SHRUB2_GM1
+        };
+
+        struct AssetStorageLabel {
+            uint8_t assetType; // Sourcehold::Assets::AssetType -> Can I somehow use AssetType instead of this uint stuff?
+            ghc::filesystem::path filename;
+        };
+
+        class AssetStorage {
+            public:
+            static map<string, shared_ptr<Game::Gm1File>> gm1Memory;
```

Comment:
In this case you want to use weak_ptr.
Take a look at GameManager, since there is already a cache implemented.

Reply:
I've to look at those std pointers, because I don't rly understand them :/
Switched to bit different architecture - I store data for systems in system class - so it will be easier I think :)